### PR TITLE
Test with both old pypy and new pypy

### PR DESCRIPTION
**Pull request by alex**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -21,7 +21,9 @@ matrix:
         - python: 3.5
           env: TOXENV=py35
         - python: pypy
-          env: TOXENV=pypy
+          env: TOXENV=pypy PYPY_VERSION=4.0.1
```
Comment (reaperhulk) [inline .travis.yml]: I believe we actually claim to support pypy 2.6, so if we want oldest/newest this should probably become 2.6?

File: docs/conf.py [hunk 2]
```diff
@@ -176,4 +176,4 @@
 
 # Retry requests in the linkcheck builder so that we're resillient against
 # transient network errors.
-linkcheck_retries = 2
+linkcheck_retries = 3
```
Comment (reaperhulk) [inline docs/conf.py]: This looks like it belongs in another PR.

**Comment by reaperhulk** [comment]
Aww damn. We need OldPyPy coverage for the non-static callback path. Maybe we need 4.0.1 for that still then?

**Comment by alex** [comment]
Let's do all 3 I guess
